### PR TITLE
fix: webhooks - don't encode URLs that already contain "%"

### PR DESCRIPTION
**Pull request by bdshadow**
Builds on #3709 (webhook URL double-encoding fix by @jdimeo). This branch cherry-picks that PR's commits unchanged and adds two things on top:

### 1. Fix the failing `WebhookAutomationTest`
#3709 switches `WebhookExecutor` from the `String` overload of `restTemplate.exchange(...)` to the `URI` overload. That broke the test in two ways:
- `mockWebhookResponse` stubbed `exchange(any<String>(), …)`, so the new `URI`-overload call never matched the stub and the mock returned `null` → NPE on `responseEntity.statusCode`.
- `verifyWebhookExecuted` asserted `callArguments[0].isEqualTo(url)`, comparing the recorded arg (now a `URI`) against a `String`.

Fixed by matching `any<URI>()` and asserting on `callArguments[0].toString()`.

### 2. Wrap the URI construction in the existing `try/catch`
In #3709 the `UriComponentsBuilder.fromUriString(...).build(...).toUri()` call sits **outside** the `try`, so a malformed/edge-case URL throws a raw `IllegalArgumentException`/`IllegalStateException` instead of `WebhookExecutionFailed`. That wrapper matters:
- the "test webhook" endpoint (`WebhookConfigService.test`) only catches `WebhookException`, so a bad URL would return HTTP 500 instead of `success = false`;
- `WebhookExecutionFailed : ExpectedUserError`, so batch/automation error classification would mis-label a user error as an internal one.

Moving the URI build inside the `try` restores both behaviors.

## Remaining concern for the maintainers (not fixed here — needs a decision)
The `fullyEncoded` heuristic can't distinguish an **intentional percent-encoded triplet** from a **literal `%` that happens to be followed by two hex digits** — the two are textually identical, so no regex can resolve it.

Example that regresses vs. `main`:
```
https://hooks.example.com/t/order%b5c9
```
`%b5` is a valid triplet → `fullyEncoded = true` → passed through untouched → the receiver percent-decodes `%b5` into byte `0xB5`, so it gets a garbled path instead of the literal `order%b5c9`. Before the change, `%` was encoded to `%25` and the literal string arrived intact.

The only robust fix is a contract decision rather than a smarter heuristic: require the stored webhook URL to always be a fully-encoded RFC-3986 URI (always `build(true)`, validate/normalize at save time, document that a literal `%` must be written `%25`). Flagging for discussion; happy to implement whichever direction the maintainers prefer.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved webhook request URL handling by validating webhook target URLs before dispatch.
  * Correctly constructs the outbound dispatch URI (including proper handling of percent-encoded characters) to ensure requests are sent to the intended endpoint.
  * Updated webhook execution verification to align with URI-based request handling while keeping existing signing and error-handling behavior.
* **Tests**
  * Adjusted webhook REST call mocking and assertions to use URI arguments consistently.

### CODE REVIEW

File: backend/data/src/main/kotlin/io/tolgee/component/automations/processors/WebhookExecutor.kt [hunk 1]
```diff
@@ -32,8 +33,14 @@ class WebhookExecutor(
     val request = HttpEntity(stringData, headers)
 
     try {
+      // Treat as already-encoded only when every '%' starts a valid percent-encoded triplet.
+      val fullyEncoded =
+        config.url.contains("%") &&
+          Regex("%(?![0-9A-Fa-f]{2})").find(config.url) == null
+      val uri = UriComponentsBuilder.fromUriString(config.url).build(fullyEncoded).toUri()
```
[the remark this answers is not in the document]
Comment (bdshadow) [reply backend/data/src/main/kotlin/io/tolgee/component/automations/processors/WebhookExecutor.kt:40]: 1. It's the same unsolvable-by-heuristic ambiguity. %b2 in a stored URL is textually identical whether it means "an encoded byte" or "a literal %b2". The build(true) vs build(false) choice can't be inferred from the string - this is the https://hooks.example.com/t/order%b5c9 case. The finding correctly concludes the runtime inference should go away, but that's a restatement of the limitation, not a bug with a local fix.
2. The proposed remedy is a contract change, not a minimal edit. "Normalize or validate URLs when saved, and have WebhookExecutor build the persisted URL using that established representation" means:
  - deciding the canonical form (require a fully-encoded RFC-3986 URI -> always build(true), and a literal % must be written %25),
  - adding save-time normalization/validation in WebhookConfigService.create/update,
  - a migration/back-compat story for existing rows,
  - and it changes behavior for users who currently store unencoded URLs.

**Review by dkrizan — APPROVED** [review]
